Packet++/src/DnsResourceData.cpp

```diff
@@ -34,7 +34,7 @@ void IDnsResourceData::encodeName(const std::string& decodedName, char* result,
 }
 
 
-StringDnsResourceData::StringDnsResourceData(const uint8_t* dataPtr, size_t dataLen, IDnsResource* dnsResource)
+StringDnsResourceData::StringDnsResourceData(const uint8_t* dataPtr, size_t, IDnsResource* dnsResource)
 {
 	char tempResult[256];
 	decodeName((const char*)dataPtr, tempResult, dnsResource);
```

Comment:
I think decodeName should have the length of the buffer in param

Reply:
`decodeName` has its own checks to make sure data is not out of bounds:
https://github.com/seladb/PcapPlusPlus/blob/5e262be0ec099f2b845eed64a5a5d8d28baf5481/Packet%2B%2B/src/DnsResource.cpp#L59

This c'tor is mostly used internally (although it's public), but maybe we can check if `dataPtr != nullptr` and `dataLen > 0`?

Reply:
Yes but I still think it's more proper to always pass a buffer with it's length and not rely on class member.
Another possibility would be to change the pointer to an offset of m_Data.

Reply:
Yes I agree, however I think we can do it in a separate PR. Decoding DNS messages is quite tricky and has a lot of edge cases, so it took me some time to get it right. `decodeName()` is a big part of it so any change in this method should be done and tested carefully

Reply:
Agree